File: pbmx_ffi/java/io/rmf/pbmx/ShiftProof.java
package io.rmf.pbmx;

import io.rmf.pbmx.ffi.LibPbmx;
import com.sun.jna.Pointer;

public final class ShiftProof {
    ShiftProof(Pointer handle) {
        this.handle = handle;
    }

    @Override
    protected void finalize() {
        LibPbmx.INSTANCE.pbmx_delete_shift_proof(this.handle);
    }

    Pointer handle;
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/ShiftProof.kt
package io.rmf.pbmx

import com.sun.jna.Pointer
import io.rmf.pbmx.ffi.FFI

class ShiftProof internal constructor(internal var handle: Pointer) {

    protected fun finalize() {
        FFI.pbmx_delete_shift_proof(this.handle)
    }
}
